ambry-cloud/src/test/java/com.github.ambry.cloud/CloudBlobStoreTest.java

```diff
@@ -614,4 +617,121 @@ private BlobId getUniqueId(short accountId, short containerId, boolean encrypted
+    hosts.put(remoteHost.dataNodeId, remoteHost);
+    MockConnectionPool connectionPool = new MockConnectionPool(hosts, clusterMap, 4);
+
+    // Generate BlobIds for following PUT.
+    short blobIdVersion = CommonTestUtils.getCurrentBlobIdVersion();
+    short accountId = Utils.getRandomShort(TestUtils.RANDOM);
+    short containerId = Utils.getRandomShort(TestUtils.RANDOM);
+    boolean toEncrypt = TestUtils.RANDOM.nextBoolean();
+    List<BlobId> blobIdList = new ArrayList<>();
+    for (int i = 0; i < 6; i++) {
+      blobIdList.add(
+          new BlobId(blobIdVersion, BlobId.BlobIdType.NATIVE, ClusterMapUtils.UNKNOWN_DATACENTER_ID, accountId,
+              containerId, partitionId, toEncrypt, BlobId.BlobDataType.DATACHUNK));
+    }
+
+    // Set up VCR
+    Properties props = new Properties();
+    setBasicProperties(props);
+    props.setProperty("clustermap.port", "12300");
+    props.setProperty("vcr.ssl.port", "12345");
+
+    ReplicationConfig replicationConfig = new ReplicationConfig(new VerifiableProperties(props));
+    ClusterMapConfig clusterMapConfig = new ClusterMapConfig(new VerifiableProperties(props));
+    CloudConfig cloudConfig = new CloudConfig(new VerifiableProperties(props));
+    CloudDataNode cloudDataNode = new CloudDataNode(cloudConfig, clusterMapConfig);
+
+    LatchBasedInMemoryCloudDestination latchBasedInMemoryCloudDestination =
+        new LatchBasedInMemoryCloudDestination(blobIdList);
+    CloudReplica cloudReplica = new CloudReplica(cloudConfig, partitionId, cloudDataNode);
+    CloudBlobStore cloudBlobStore =
+        new CloudBlobStore(new VerifiableProperties(props), partitionId, latchBasedInMemoryCloudDestination, clusterMap,
+            new VcrMetrics(new MetricRegistry()));
+    cloudBlobStore.start();
+
+    // Create ReplicaThread and add RemoteReplicaInfo to it.
+    ReplicationMetrics replicationMetrics = new ReplicationMetrics(new MetricRegistry(), Collections.emptyList());
+    ReplicaThread replicaThread =
+        new ReplicaThread("threadtest", new MockFindToken.MockFindTokenFactory(), clusterMap, new AtomicInteger(0),
+            cloudDataNode, connectionPool, replicationConfig, replicationMetrics, null, storeKeyConverter, transformer,
+            clusterMap.getMetricRegistry(), false, cloudDataNode.getDatacenterName(), new ResponseHandler(clusterMap),
+            new MockTime());
+
+    for (ReplicaId replica : partitionId.getReplicaIds()) {
+      if (replica.getDataNodeId() == remoteHost.dataNodeId) {
+        RemoteReplicaInfo remoteReplicaInfo =
+            new RemoteReplicaInfo(replica, cloudReplica, cloudBlobStore, new MockFindToken(0, 0), Long.MAX_VALUE,
+                SystemTime.getInstance(), new Port(remoteHost.dataNodeId.getPort(), PortType.PLAINTEXT));
+        replicaThread.addRemoteReplicaInfo(remoteReplicaInfo);
+        break;
+      }
+    }
+
+    EnumSet<StoreGetOptions> getOptions = EnumSet.noneOf(StoreGetOptions.class);
```

Comment:
Please also test the other options and verify you get the correct error code.

Reply:
fixed.